### ISSUE TITLE
How to use in a react-stack..?

### ISSUE THREAD

**Issue by rvetere** [selo796/find-unused-css#2]
Hi there, looks very promising!

I just don't see how we should define all of the hundrets of css files we got trough our component-based developement workflow.

What is missing for my taste is a regex possibility to define the path for my css-file -> like globbing i should be able to say `/app/**/*.css` just to dynamically grab _all_ of them in my project.

The same problem with html - it is not html at all in our stack. The html in charge would be found as JSX in normal ".js" files. Well i hope the package doesn't care if it is true html i give him - but still, the same problem here with missing globbing as far as i can see ;-)

Hard to solve such a thing? If you point me in the direction where this should take place i can have a look in the code and introduce a PR if you want - we would really like to use such a thing in our Stack! ;-)

**Comment by selo796** [comment]
Hi Remo,

these two functionalities are still missing. I will have a look and let you inform for that. I think "Dynamically grabing" of css files are easy to fix but the JSX files are not. 

I also want to support Angular2, React and other frameworks but it will take same times :). 

Regards
Selcuk

**Comment by selo796** [comment]
Hi Remo,

i need a parser for JSX files which should return all the class names from the file. In React or AngularJS you can apply class names also dynamically. Therefore it is not so easy to solve that issue. Let me please know if you have any idea.

**Comment by rvetere** [comment]
mhm... at least in our stack we're using heavily and only "css-modules"..... let me think of that a little bit longer -> because the webpack process knows verry well _all_ the used classes which comes out of the JSX in the end of the build, just by processing css-modules!

So, i'm pretty sure we can make use of this existing functionality somehow and this would fix it right away - well, only if you use css-modules i guess but a totally bleeding-edge stack should use this anyway, also angular2 😉

**Comment by selo796** [comment]
This first one is already done: 

- _find out all the existing css defintions of the project -> just by going trough all *.css files and maybe regex/parse out the definitions from there (not too hard to parse a css - maybe we can even take phantomJS, let him render a dummy page with the css included and just ask his DOM engine which definitions he sees - BAM)_

The second point is not so easy:

1- You can have some selectors e.g id, class, attributes etc. in html files. (id and class selectors are already done)
2- parsing js files are depending on the library/framework. e.g react uses classNames, angular2 uses styles, jquery uses appendClass etc... Beside of this there is also adding dynamically class some think like this: 

```html
<div className="btn-group pull-right {this.props.showBulkActions ? 'show' : 'hidden'}">
```
Therefore we have to know for which file/library we want to analyze.

**Comment by selo796** [comment]
Hi Remo,

i even published a new version. I am very thankfully for further feedbacks.

Regards
Selcuk
